pkg/frontend/mysql_protocol.go

```diff
@@ -2706,43 +2706,29 @@ func (mp *MysqlProtocolImpl) MakeEOFPayload(warnings, status uint16) []byte {
 
 // receiveExtraInfo tries to receive salt and labels read from proxy module.
 func (mp *MysqlProtocolImpl) receiveExtraInfo(rs goetty.IOSession) {
-	saltLen := 20
 	// TODO(volgariver6): when proxy is stable, remove this deadline setting.
 	if err := rs.RawConn().SetReadDeadline(time.Now().Add(defaultSaltReadTimeout)); err != nil {
 		logDebugf(mp.GetDebugString(), "failed to set deadline for salt updating: %v", err)
 		return
 	}
-	data := make([]byte, saltLen)
-	n, err := rs.RawConn().Read(data)
-	if err != nil {
-		// Something wrong when try to read the salt value.
-		// If the error is timeout, we treat it as normal case and do not update salt.
-		//
-		// TODO(volgariver6): we should change the port of the internal execution from
-		// 6001 to the proxy listen port.
-		if err, ok := err.(net.Error); !ok || err.Timeout() {
-			logError(mp.ses, mp.GetDebugString(),
-				"Failed to get salt",
-				zap.Error(err))
-		}
-	} else if n != saltLen {
-		logError(mp.ses, mp.GetDebugString(),
-			"Failed to get salt",
-			zap.Error(err))
-	} else {
-		mp.SetSalt(data)
-	}
-
-	// Read requested labels from proxy.
-	label := &proxy.RequestLabel{}
+	extraInfo := &proxy.ExtraInfo{}
 	reader := bufio.NewReader(rs.RawConn())
-	if err = label.Decode(reader); err != nil {
-		logError(mp.ses, mp.GetDebugString(),
-			"Failed to get CN labels",
-			zap.Error(err))
+	if err := extraInfo.Decode(reader); err != nil {
+		if err != nil {
+			// Something wrong when try to read the salt value.
+			// If the error is timeout, we treat it as normal case and do not update salt.
+			if err, ok := err.(net.Error); ok && err.Timeout() {
+				logInfo(mp.ses, mp.GetDebugString(), "cannot get salt, maybe not use proxy",
+					zap.Error(err))
+			} else {
+				logError(mp.ses, mp.GetDebugString(), "failed to get extra info",
+					zap.Error(err))
+			}
+		}
 	} else {
-		mp.GetSession().requestLabel = label.Labels
-		logDebugf(mp.GetDebugString(), "got requested CN labels: %v", *label)
+		mp.SetSalt(extraInfo.Salt)
+		mp.GetSession().requestLabel = extraInfo.Label.Labels
+		mp.GetSession().internalConn = extraInfo.InternalConn
```

Comment:
考虑到cn 可以在没有 proxy的情况下运行，建议改为: 0, 1, 2, ...。默认值对应unset
```suggestion
if extraInfo.InternalConn {
    mp.GetSession().connType = 1; // 0: unset; 1: private networks; 2: others
}
```

Reply:
done